packages/back-end/src/integrations/SqlIntegration.ts

```diff
@@ -3126,10 +3218,117 @@ AND event_name = '${eventName}'`,
       GROUP BY ${groupByColumn || eventColumn}
     `;
   }
-  async getEventsTrackedByDatasource(
-    schemaFormat: AutoMetricSchemas,
+
+  async getAutoFactTablesToCreate(
+    schemaFormat: SchemasThatSupportAutoFactTablesAndMetrics,
+    existingFactTables: FactTableInterface[],
+    datasourceId: string,
+    schema?: string
+  ): Promise<AutoFactTableToCreate[]> {
+    // GA4 stores all events in on the events_* table(s) - so we don't need to get a list of all events
```

Comment:
The only downside of creating a single Fact Table per GA4 Data source (code wise) is that we have to have this messy block. Open to ideas on how to approach this differently.